packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -666,65 +674,106 @@ describe('KeyringController', () => {
-    );
-  });
+      const signedTx = await keyringController.signTransaction(
+        unsignedEthTx,
+        account,
+      );
+      expect(signedTx.v).not.toBeUndefined();
+      expect(signedTx).not.toBe('');
+    });
 
-  /**
-   * Task added to improve check for valid transaction in signTransaction method
-   * https://github.com/MetaMask/core/issues/800
-   */
-  it('should not sign transaction if transaction is not valid', async () => {
-    await expect(async () => {
-      const account = initialState.keyrings[0].accounts[0];
-      await keyringController.signTransaction({}, account);
-    }).rejects.toThrow('tx.sign is not a function');
+    it('should not sign transaction if from account is not provided', async () => {
+      await expect(async () => {
+        const account = initialState.keyrings[0].accounts[0];
+        const txParams = {
+          chainId: 5,
+          data: '0x1',
+          from: account,
+          gasLimit: '0x5108',
+          gasPrice: '0x5108',
+          to: '0x51253087e6f8358b5f10c0a94315d69db3357859',
+          value: '0x5208',
+        };
+        const unsignedEthTx = TransactionFactory.fromTxData(txParams, {
+          common: new Common(commonConfig),
+          freeze: false,
+        });
+        expect(unsignedEthTx.v).toBeUndefined();
+        await keyringController.signTransaction(unsignedEthTx, '');
+      }).rejects.toThrow(
+        'No keyring found for the requested account. Error info: The address passed in is invalid/empty',
+      );
+    });
+
+    /**
+     * Task added to improve check for valid transaction in signTransaction method
+     * https://github.com/MetaMask/core/issues/800
+     */
+    it('should not sign transaction if transaction is not valid', async () => {
+      await expect(async () => {
+        const account = initialState.keyrings[0].accounts[0];
+        await keyringController.signTransaction({}, account);
+      }).rejects.toThrow('tx.sign is not a function');
+    });
   });
 
-  it('should submit password and decrypt', async () => {
-    const state = await keyringController.submitPassword(password);
-    expect(state).toStrictEqual(initialState);
+  describe('submitPassword', () => {
+    it('should submit password and decrypt', async () => {
+      const state = await keyringController.submitPassword(password);
+      expect(state).toStrictEqual(initialState);
+    });
   });
 
-  it('should subscribe and unsubscribe', async () => {
-    const listener = sinon.stub();
-    keyringController.subscribe(listener);
-    await keyringController.importAccountWithStrategy(
-      AccountImportStrategy.privateKey,
-      [privateKey],
-    );
-    expect(listener.called).toBe(true);
-    keyringController.unsubscribe(listener);
-    await keyringController.removeAccount(
-      '0x51253087e6f8358b5f10c0a94315d69db3357859',
-    );
-    expect(listener.calledTwice).toBe(false);
+  describe('subscribe', () => {
```

Comment:
Nit: I know it doesn't exactly follow our usual conventions, but maybe we can mention both methods here since they're both under-test in this test.
```suggestion
  describe('subscribe and unsubscribe', () => {
```